### ISSUE TITLE
 Constructor overriding should be smarter 

### ISSUE THREAD

**Issue by feinstein** [dart-lang/sdk#39944]
Opened [here](https://youtrack.jetbrains.com/issue/WEB-43108) as:

I extended a class and IntelliJ suggested me I should override a constructor:

![image](https://user-images.githubusercontent.com/8120472/71550531-95f47e80-29b1-11ea-9cd1-d3d378bcb398.png)

But the automatic override using `Alt+Enter`  generates a constructor with an error:

![image](https://user-images.githubusercontent.com/8120472/71550535-a3aa0400-29b1-11ea-8ce6-43bb1885501e.png)

I don't know how difficult is, but it would be better if this tool didn't generate something with an error.

**Comment by devoncarew** [comment]
Thanks for the report; can you provide some more details about the FirebaseAuth class - I assume that it doesn't have a public generative constructor - just a public factory constructor?

**Comment by feinstein** [comment]
Yes, it can be found at the Flutter Fire Repo, since I am on my phone I will just paste it here:

https://github.com/FirebaseExtended/flutterfire/blob/master/packages/firebase_auth/firebase_auth/lib/src/firebase_auth.dart

```dart
class FirebaseAuth {
  FirebaseAuth._(this.app);

  /// Provides an instance of this class corresponding to `app`.
  factory FirebaseAuth.fromApp(FirebaseApp app) {
    assert(app != null);
    return FirebaseAuth._(app);
  }

  /// Provides an instance of this class corresponding to the default app.
  static final FirebaseAuth instance = FirebaseAuth._(FirebaseApp.instance);

  final FirebaseApp app;

... 
```

**Comment by natebosch** [comment]
To clarify, with this definition you can't write a class which `extends FirebaseAuth`. It would definitely be nice if the analyzer made this more clear.
